[docs] Update billing docs (#37496)
expo/expo: docs/pages/billing/invoices-and-receipts.mdx

To download and view an invoice for a billing period, go to [Receipts](https://expo.dev/settings/receipts) and:

> - Click the **Date** for the billing period corresponding to the invoice. You will be navigated to a page hosted by Stripe. As an example, the March 22, 2025 invoice below links to this page:

Reviewer: The date in this example is consistent with the date in the image below. If we want to update the date here, we should update the image below and in the next section.

To download and view a receipt for a billing period, go to [Receipts](https://expo.dev/settings/receipts) and:

> - Click the **Date** for the billing period corresponding to the receipt. You will be navigated to the appropriate receipt hosted by Stripe. As an example, the March 22, 2025 receipt below links to this page:

Reviewer: Same as the previous one.

### Starter charges

In the third example, the subscriber is on a Starter plan. Any charges incurred during the billing period are listed in the invoice:

| Description                                                         | Quantity | Unit price |     Amount |
| ------------------------------------------------------------------- | -------: | ---------: | ---------: |
| _FEB 1 - MAR 1, 2025_                                               |          |            |            |
| EAS Build - Build (Android: 10 medium builds; iOS 10 medium builds) |        1 |     $30.00 |     $30.00 |
| EAS Build - Plan credit                                             |        1 |    -$30.00 |    -$30.00 |
| _MAR 1 - APR 1, 2025_                                               |          |            |            |
| Expo Application Services - Starter                                 |        1 |     $19.00 |     $19.00 |
| **Total (USD)**                                                     |          |            | **$19.00** |

In the above example:

> - The first line item describes the EAS Build usage for the billing period of February 1 to March 1, 2025. It contains all the details about how many Android and iOS builds were created during this billing period and their cost.

Reviewer: ```suggestion
- The first line item describes the EAS Build usage for the billing period of February 1 to March 1, 2025. It contains all the details about the number of Android and iOS builds created during this billing period and their cost.
```

I'd suggest to improve this line for clear readability.
